### PR TITLE
Add TestSystem to testkit

### LINKED ISSUE #622: Add TestSystem to `testkit` package
Modeled after `TestClock`, the `TestSystem` should be an implementation of `System.Service` that can be constructed with a `Ref[TestSystem.Data]`, which should read property variables and associated data from the test data structure. For convenience, a version could be provided that takes a raw `TestSystem.Data`, which is not inside a `Ref`, and would be ideal for static data.

### PR DESCRIPTION
**Pull request by bhop**
fixes #622

### CODE REVIEW

File: testkit/jvm/src/main/scala/scalaz/zio/testkit/TestSystem.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+object TestSystem {
+
+  def apply(data: Data): UIO[TestSystem] =
+    Ref.make(data).map(TestSystem(_))
+
+  case class Data(
+    properties: Map[String, String] = Map.empty,
+    envs: Map[String, String] = Map.empty,
+    lineSeparator: String = ";"
```
Comment (jdegoes) [inline testkit/jvm/src/main/scala/scalaz/zio/testkit/TestSystem.scala]: This should default to "\n"

**Comment by bhop** [comment]
Thanks @jdegoes. I updated pull request. Travis build passed too.

**Comment by jdegoes** [comment]
giphy

**Comment by jdegoes** [comment]
@bhop Congratulations on being a ZIO contributor! Please stick around. 😄

**Comment by bhop** [comment]
@jdegoes Thanks!

**Pull request merged**